modules/core/src/main/scala/org/scalasteward/core/application/Context.scala

```diff
@@ -94,22 +117,46 @@ object Context {
         ClientConfiguration.disableFollowRedirect[F],
         middleware
       )
-      fileAlg = FileAlg.create(logger, F)
-      processAlg = ProcessAlg.create(config.processCfg)(logger, F)
-      workspaceAlg = WorkspaceAlg.create(config)(fileAlg, logger, F)
-      context <- Resource.eval(
-        step1(config)(
-          defaultClient,
-          UrlCheckerClient(urlCheckerClient),
-          fileAlg,
-          logger,
-          processAlg,
-          workspaceAlg,
-          F
-        )
-      )
+      fileAlg0 = FileAlg.create(logger0, asyncF)
+      context <- usage match {
+        case StewardUsage.Regular(config) =>
+          initRegular(config)(
+            defaultClient,
+            UrlCheckerClient(urlCheckerClient),
+            fileAlg0,
+            logger0,
+            asyncF
+          ).map(StewardContext.Regular(_))
+
+        case StewardUsage.ValidateRepoConfig(file) =>
+          implicit val fileAlg: FileAlg[F] = fileAlg0
+          implicit val logger: Logger[F] = logger0
+          Resource.pure[F, StewardContext[F]](initValidateRepoConfig(file))
+      }
+
     } yield context
 
+  def initRegular[F[_]](config: Config)(implicit
+      client: Client[F],
+      urlCheckerClient: UrlCheckerClient[F],
+      fileAlg: FileAlg[F],
+      logger: Logger[F],
+      asyncF: Async[F]
+  ): Resource[F, Context[F]] = {
+    implicit val processAlg = ProcessAlg.create(config.processCfg)
+    implicit val workspaceAlg = WorkspaceAlg.create(config)
+    Resource.eval(step1(config))
+  }
+
+  def initValidateRepoConfig[F[_]](file: File)(implicit
+      fileAlg: FileAlg[F],
+      logger: Logger[F],
+      asyncF: MonadThrow[F]
```

Comment:
```suggestion
      F: MonadThrow[F]
```
Naming convention in this repo.